ts/Extensions/Annotations/Popup.ts

```diff
@@ -865,143 +944,394 @@ H.Popup.prototype = {
                 'add'
             );
-
             buttonParentDiv = tabsContainers[0]
                 .querySelectorAll('.' + PREFIX + 'popup-rhs-col')[0];
 
             this.addButton(
-                buttonParentDiv as any,
+                buttonParentDiv as HTMLDOMElement,
                 lang.addButton || 'add',
                 'add',
-                callback,
-                buttonParentDiv as any
+                buttonParentDiv as HTMLDOMElement,
+                callback
             );
 
             // EDIT tab
-            this.addColsContainer(tabsContainers[1] as any);
+            this.addColsContainer(tabsContainers[1] as HTMLDOMElement);
             indicators.addIndicatorList.call(
                 this,
                 chart,
-                tabsContainers[1] as any,
+                tabsContainers[1] as HTMLDOMElement,
                 'edit'
             );
 
             buttonParentDiv = tabsContainers[1]
                 .querySelectorAll('.' + PREFIX + 'popup-rhs-col')[0];
 
             this.addButton(
-                buttonParentDiv as any,
+                buttonParentDiv as HTMLDOMElement,
                 lang.saveButton || 'save',
                 'edit',
-                callback,
-                buttonParentDiv as any
+                buttonParentDiv as HTMLDOMElement,
+                callback
             );
             this.addButton(
-                buttonParentDiv as any,
+                buttonParentDiv as HTMLDOMElement,
                 lang.removeButton || 'remove',
                 'remove',
-                callback,
-                buttonParentDiv as any
+                buttonParentDiv as HTMLDOMElement,
+                callback
             );
         },
+
+        /**
+         * Filter object of series which are not indicators.
+         * If the filter string exists, check against it.
+         *
+         * @private
+         *
+         * @param {Highcharts.FilteredSeries} series
+         *        All series are available in the plotOptions.
+         *
+         * @param {string|undefined} filter
+         *        Applied filter string from the input.
+         *        For the first iteration, it's an empty string.
+         *
+         * @return {Array<Highcharts.FilteredSeries>} filteredSeriesArray
+         *         Returns array of filtered series based on filter string.
+         */
+        filterSeries: function (
+            this: Highcharts.Popup,
+            series: Series,
+            filter?: string
+        ): [Highcharts.FilteredSeries] {
+            const popup = this,
+                indicators = popup.indicators,
+                lang = popup.chart && popup.chart.options.lang,
+                indicatorAliases = lang &&
+                    lang.navigation &&
+                    lang.navigation.popup &&
+                    lang.navigation.popup.indicatorAliases;
+            let filteredSeriesArray: [Highcharts.FilteredSeries] = [] as any,
```

Comment:
Array type should be Array<Highcharts.FilteredSeries>, then you can remove `any`.